dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java

```diff
@@ -17,46 +22,148 @@
 import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
 import com.dotmarketing.portlets.workflows.business.SystemWorkflowConstants;
 import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
 import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
 import com.dotmarketing.tag.model.TagInode;
+import com.liferay.portal.model.User;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.jetbrains.annotations.NotNull;
 import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
+@RunWith(DataProviderRunner.class)
 public class SaveContentActionletTest extends BaseWorkflowIntegrationTest {
 
-    private static WorkflowAPI workflowAPI = null;
-    private static ContentTypeAPI contentTypeAPI = null;
-    private static ContentType customContentType = null;
-    private static final int LIMIT = 20;
 
-    @BeforeClass
-    public static void prepare() throws Exception {
-        //Setting web app environment
+    private static User systemUser = APILocator.systemUser();
+    private static List<ContentType> contentTypes = new ArrayList<>();
 
-        IntegrationTestInitService.getInstance().init();
-        workflowAPI = APILocator.getWorkflowAPI();
-        contentTypeAPI = APILocator.getContentTypeAPI(APILocator.systemUser());
+    private static class TestCase {
+        private final boolean respectFrontendRoles;
+        private final User user;
+        private final boolean hasSaveActionPermission;
+        private final boolean hasContentTypeAddChildrenPermission;
+        private final ContentType contentType;
+
+        private TestCase(
+                final boolean respectFrontendRoles,
+                final User user,
+                final ContentType contentType,
+                final boolean hasSaveActionPermission,
+                final boolean hasContentTypeAddChildrenPermission) {
+            this.respectFrontendRoles = respectFrontendRoles;
+            this.user = user;
+            this.hasSaveActionPermission = hasSaveActionPermission;
+            this.hasContentTypeAddChildrenPermission = hasContentTypeAddChildrenPermission;
+            this.contentType = contentType;
+        }
+    }
+
+    @DataProvider
+    public static Object[] usersAndContentTypeWithoutHostField() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)